qcodes/tests/drivers/test_ami430.py

```diff
@@ -440,6 +555,230 @@ def test_blocking_ramp_parameter(current_driver, caplog):
+    np.testing.assert_almost_equal(
+        field_ramp_limit, current_ramp_limit*coil_constant)
+    np.testing.assert_almost_equal(
+        field_limit, current_limit*coil_constant)
+
+
+def test_current_and_field_params_interlink__change_coil_constant(
+        ami430, factor=3):
+    """
+    Test that after changing ``change_coil_constant``, the values of the
+    ``current_*`` parameters remain the same while the values of the
+    ``field_*`` parameters change proportionally. At the end just ensure that
+    the values of the ``coil_constant``-dependent parameters are correctly
+    proportional to each other.
+    """
+    coil_constant_old = ami430.coil_constant()
+    current_ramp_limit_old = ami430.current_ramp_limit()
+    field_ramp_limit_old = ami430.field_ramp_limit()
+    current_limit_old = ami430.current_limit()
+    field_limit_old = ami430.field_limit()
+
+    coil_constant_new = coil_constant_old * factor
+
+    ami430.coil_constant(coil_constant_new)
+
+    current_ramp_limit_new_expected = current_ramp_limit_old
+    current_limit_new_expected = current_limit_old
+    field_ramp_limit_new_expected = field_ramp_limit_old * factor
+    field_limit_new_expected = field_limit_old * factor
+
+    current_ramp_limit = ami430.current_ramp_limit()
+    field_ramp_limit = ami430.field_ramp_limit()
+    coil_constant = ami430.coil_constant()
+    current_limit = ami430.current_limit()
+    field_limit = ami430.field_limit()
+
+    # The following parameters are expected to change
+    np.testing.assert_almost_equal(
+        coil_constant, coil_constant_new)
+    np.testing.assert_almost_equal(
+        current_ramp_limit, current_ramp_limit_new_expected)
+    np.testing.assert_almost_equal(
+        field_ramp_limit, field_ramp_limit_new_expected)
+    np.testing.assert_almost_equal(
+        current_limit, current_limit_new_expected)
+    np.testing.assert_almost_equal(
+        field_limit, field_limit_new_expected)
+
+    # Proportions are expected to hold between field and current parameters
+    np.testing.assert_almost_equal(
+        field_ramp_limit, current_ramp_limit*coil_constant)
+    np.testing.assert_almost_equal(
+        field_limit, current_limit*coil_constant)
+
+
+def test_current_and_field_params_interlink__permutations_of_tests(ami430):
+    """
+    As per one of the user's request, the
+    test_current_and_field_params_interlink__* tests are executed here with
+    arbitrary 'factor's and with all permutations. This test ensures the
+    robustness of the driver even more.
+
+    Note that the 'factor's are randomized "manually" because of the
+    possibility to hit the limits of the parameters.
+    """
+    with warnings.catch_warnings():
+        # this is to avoid AMI430Warning about "maximum ramp rate", which
+        # may show up but is not relevant to this test
+        warnings.simplefilter('ignore', category=AMI430Warning)
+
+        test_current_and_field_params_interlink_at_init(ami430)
+
```

Comment:
Would it make sense to make this a bit more DRY by looping over 3 tuples or pytest parametrize ?

Reply:
sure. i can probably hard-code some values and use permutations from itertools or smth.

but only if you insist :)

Reply:
@jenshnielsen ? :)

Reply:
I would prefer but I don't insist. That's why its a question :)

Reply:
ok :)